torch/csrc/jit/codegen/cuda/expr_evaluator.cpp

```diff
@@ -38,11 +38,14 @@ void EvaluationContext::print() const {
 }
 
 c10::optional<Int::ScalarType> ExpressionEvaluator::evaluate(
-    Val* val,
+    const Val* val,
     const EvaluationContext* context) {
   TORCH_CHECK(context != nullptr);
   ExpressionEvaluator evaluator(context);
-  evaluator.traverseFrom(context->fusion(), {val}, false);
+  // IterVisitor only supports Val*, so we need to strip away const from
+  // our target val. This is ugly, but it's ok since we know that all
+  // the original Fusion nodes are allocated from the heap as non-const
+  evaluator.traverseFrom(context->fusion(), {const_cast<Val*>(val)}, false);
```

Comment:
lo and behold, clang-tidy's putting the foot down, and the CI check treats clang-tidy warnings as the law.

This means that this simple workaround is no longer an option.